src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RangeAttribute.cs

```diff
@@ -70,6 +70,21 @@ public RangeAttribute(Type type, string minimum, string maximum)
         /// </summary>
         public Type OperandType { get; }
 
+        /// <summary>
+        /// Determines whether string values for <see cref="Minimum"/> and <see cref="Maximum"/> are parsed in the invariant
+        /// culture rather than the current culture in effect at the time of the validation.
+        /// </summary>
+        public bool ParseLimitsInInvariantCulture { get; set; }
+
+        /// <summary>
+        /// Determines whether any conversions necessary from the value being validated to <see cref="OperandType"/> as set
+        /// by the <c>type</c> parameter of the <see cref="RangeAttribute(Type, string, string)"/> constructor are carried
+        /// out in the invariant culture rather than the current culture in effect at the time of the validation.
+        /// </summary>
+        /// <remarks>This property has no effects with the constructors with <see cref="int"/> or <see cref="double"/>
+        /// parameters, for which the invariant culture is always used for any conversions of the validated value.</remarks>
+        public bool ConvertValueInInvariantCulture { get; set; }
```

Comment:
@JonHanna @ajcvickers 

I may be missing something important, but I wonder if we really need two new separate flags instead of just a single `UseInvariantCulture` (or similar) flag that controls both parsing and any conversions. 

I suspect that customers that RangeAttribute customers that need it to work with invariant culture would be able to use that, and having two flags seems to make things harder.

Reply:
I can't really think of a reason for `ParseLimitsInInvariantCulture` being false apart from backwards compatibility, since the value would be set in source code, but it seems much more reasonable to parse input values in the current locale.

Reply:
@divega I'm okay with two flags. Is this something you want to follow up on more?

Reply:
@JonHanna @ajcvickers after re-understanding the issue I agree we can add two separate flags. It was a matter of turning off everything else to focus on this for a few minutes. Sorry it took me so long.